lib/index.js

```diff
@@ -48,6 +48,36 @@ function remove(tracks) {
     return TrackPlayer.remove(tracks);
 }
 
+function registerEventHandler(handler) {
+    if (Platform.OS !== 'android') {
+        function warpResponse(event, payload) {
+            // transform into headlessTask format and return to handler
+            const additionalKeys = payload || {};
+            handler({ type: event, ...additionalKeys });
```

Comment:
"handler" is a function that returns the event handler function. It's looks unnecessary, but that's how react native works

```js
AppRegistry.registerHeadlessTask('TrackPlayer', () => {
    return async (data) {
        // Handle event
    }
});
```

I guess it's to only initialize the event handler code when it needs to.
https://facebook.github.io/react-native/docs/headless-js-android.html#the-js-api